src/spaces/utilities.jl

```diff
@@ -176,46 +200,21 @@ Example usage in [Battle Royale](
 function walk!(
     agent::AbstractAgent,
     direction::NTuple{D,Int},
-    model::ABM{<:AbstractGridSpace{D,true}};
-    kwargs...,
+    model::ABM{<:AbstractGridSpace};
+    ifempty::Bool = true
 ) where {D}
-    target = mod1.(agent.pos .+ direction, size(model.space))
-    walk_if_empty!(agent, target, model; kwargs...)
-end
-function walk!(
-    agent::AbstractAgent,
-    direction::NTuple{D,Int},
-    model::ABM{<:AbstractGridSpace{D,false}};
-    kwargs...,
-) where {D}
-    target = min.(max.(agent.pos .+ direction, 1), size(model.space))
-    walk_if_empty!(agent, target, model; kwargs...)
-end
-function walk_if_empty!(agent, target, model; ifempty::Bool = false)
-    if ifempty
-        isempty(target, model) && move_agent!(agent, target, model)
-    else
+    target = normalize_position(agent.pos .+ direction, model)
+    if !ifempty || isempty(ids_in_position(target, model))
         move_agent!(agent, target, model)
     end
 end
 
-# Continuous
-function walk!(
-    agent::AbstractAgent,
-    direction::NTuple{D,Float64},
-    model::ABM{<:ContinuousSpace{D,true}};
-    kwargs...,
-) where {D}
-    target = mod1.(agent.pos .+ direction, spacesize(model))
-    target = min.(target, prevfloat.(spacesize(model)))
-    move_agent!(agent, target, model)
-end
 function walk!(
     agent::AbstractAgent,
     direction::NTuple{D,Float64},
-    model::ABM{<:ContinuousSpace{D,false}}
+    model::ABM{<:ContinuousSpace}
 ) where {D}
-    target = min.(max.(agent.pos .+ direction, 0.0), prevfloat.(spacesize(model)))
```

Comment:
Okay, so instead of the `min, max` approach, you are using `clamp`. Have you tested which version is faster?

Reply:
Clamp, by almost 2x

Reply:
```julia
julia> a = (1.3, 2.5)
julia> b = (1.2, 2.3)
julia> f1(a,b) = clamp.(a, 0., b)
julia> f2(a,b) = min.(max.(a, 0.), b)

julia> @benchmark f1($a, $b)
BenchmarkTools.Trial: 10000 samples with 1000 evaluations.
 Range (min … max):  1.557 ns … 6.759 ns  ┊ GC (min … max): 0.00% … 0.00%
 Time  (median):     1.566 ns             ┊ GC (median):    0.00%
 Time  (mean ± σ):   1.591 ns ± 0.136 ns  ┊ GC (mean ± σ):  0.00% ± 0.00%

  █▇▃  ▁                                                    ▂
  ███▅▃█▇▃▆▄▄▁▃▃▃▃▁▁▁▁▁▁▃▃▁▁▁▁▁▁▁▁▁▄▅▄▅▇▆▇▇▇▇▆▇▇▇▇▇▇▇▆▇██▇▇ █
  1.56 ns     Histogram: log(frequency) by time     2.07 ns <

 Memory estimate: 0 bytes, allocs estimate: 0.

julia> @benchmark f2($a, $b)
BenchmarkTools.Trial: 10000 samples with 1000 evaluations.
 Range (min … max):  2.279 ns … 42.178 ns  ┊ GC (min … max): 0.00% … 0.00%
 Time  (median):     2.292 ns              ┊ GC (median):    0.00%
 Time  (mean ± σ):   2.454 ns ±  1.387 ns  ┊ GC (mean ± σ):  0.00% ± 0.00%

  █▅▄                                                        ▁
  ████▆▄▅▄▃▁▃▃▃▃▃▁▄▄▃▃▃▁▁▄▅▃▄▅▄▅▅▅▆▆▅▆▅▅▆▆▆▆▆▆▅▆▆▆▆▆▆▆▆▆▅▆▆▆ █
  2.28 ns      Histogram: log(frequency) by time     4.39 ns <

 Memory estimate: 0 bytes, allocs estimate: 0.
```